cmd/podman/images/scp.go

```diff
@@ -0,0 +1,330 @@
+// serv is an array of service destinations
+func parseLocation(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	switch len(args) {
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) { // if an image is specified, this mean we are loading from our client
+				scpOpts.Connections = append(scpOpts.Connections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+				} else { // else its a load save (order of args)
+					scpOpts.Connections = append(scpOpts.Connections, args[1])
+					scpOpts.Connections = append(scpOpts.Connections, args[0])
+				}
+			}
+		} else if strings.Contains(args[1], "::") && len(scpOpts.Connections) < 2 { // if we are given image host::
+			scpOpts.ToRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[1])
+		}
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			scpOpts.Connections = append(scpOpts.Connections, args[0])
+		}
+	}
+
+	for i, val := range scpOpts.Connections {
+		splitEnv := strings.SplitN(val, "::", 2)
+		switch len(splitEnv) {
+		case 1:
+			scpOpts.Connections[i] = splitEnv[0]
+		case 2:
+			scpOpts.ImageName = splitEnv[1]
+			//TODO: actually use the new name given by the user
+			scpOpts.Connections[i] = splitEnv[0]
+		}
+		conn, found := cfg.Engine.ServiceDestinations[scpOpts.Connections[i]]
+		if found {
+			iden = append(iden, conn.Identity)
+			url = append(url, conn.URI)
+		}
+		if !found { // no match, warn user and do a manual connection.
+			url = append(url, "ssh://"+scpOpts.Connections[i])
+			iden = append(iden, "")
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+	}
+	if len(scpOpts.Connections) == 0 { // we default to remote or whatever the default connection is
+		if len(cfg.Engine.ServiceDestinations) == 0 && strings.Contains(args[0], "::") {
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		} else {
+			connStr := cfg.Engine.ActiveService
+			if connStr == "" {
+				iden = append(iden, "")
+				url = append(url, " ")
+			}
+			if len(cfg.Engine.ServiceDestinations) == 0 && strings.Contains(args[0], "::") {
+				logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
```

Comment:
Why is it necessary to treat `len(cfg.Engine.ServiceDestnations) == 0` as a special case? Shouldn’t a warning on `!found` below exist, and automatically handle the `len(…) == 0` case?

Reply:
resolved due to no remote code